### PR TITLE
Weather UI

### PR DESCRIPTION
**Pull request by gedefaws**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

updated the weather ui which includes two recycler views and two additional layouts (one for current weather which sits on top of the main weather fragment and another one for the search which includes two buttons zipcode search and map to be implemented later). Also, includes all the components required.

Fixes # (issue)
#54 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

run on android studio

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [X] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherData.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+public class WeatherData implements Serializable {
+    private  String mTime;
+    private  String mCity;
+    private  Double mTemperature;
+    private  String mCondition;
+    private  Double mMinTemp;
+    private  Double mMaxTemp;
+    private  String mDate;
+
+    /**
```
Comment (gtullio12) [inline app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherData.java:14]: You can remove java doc for constructors

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherData.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     * constructer for the hourly weather data
+     * @param time
+     * @param temperature
+     * @param condition
+     */
+    public WeatherData(String time, Double temperature, String condition) {
+        this.mTime = time;
+        this.mTemperature = temperature;
+        this.mCondition = condition;
+
```
Comment (gtullio12) [inline app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherData.java:36]: Make sure to remove unnecessary lines for style purposes

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherListFragment.java [hunk 3]
```diff
@@ -39,5 +60,10 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
                 Navigation.findNavController(getView()).navigate(
                         WeatherListFragmentDirections.actionNavigationWeatherToWeatherMapFragment())
         );
+
```
Comment (gtullio12) [inline app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherListFragment.java:63]: You can remove these lines

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherDailyRecyclerViewAdapter.java [hunk 4]
```diff
@@ -61,13 +61,20 @@ public WeatherViewHolder(View view) {
...
         }
 
-
         /**
-         * Helper used to determine if the preview should be displayed or not.
+         * helps to set the components of the dailyweather
+         * @param weatherDaily
          */
+        void setWeather(final WeatherData weatherDaily)
+        {
```
Comment (gtullio12) [inline app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherDailyRecyclerViewAdapter.java:69]: Make sure to have the brace on the same line as the function declaration

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherDailyRecyclerViewAdapter.java [hunk 5]
```diff
@@ -61,13 +61,20 @@ public WeatherViewHolder(View view) {
...
+            mWeather = weatherDaily;
+            mWeather = weatherDaily;
+            binding.textMinTemp.setText(mWeather.getmMinTemperature());
+            binding.textMaxTemp.setText(mWeather.getmMaxTemperature());
+            binding.textCondition.setText(mWeather.getmCondition());
+            binding.textWeatherDate.setText(mWeather.getmDate());
+            //set the image icon to be implemented using if else depending on the condition
```
Comment (gtullio12) [inline app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherDailyRecyclerViewAdapter.java:76]: Is this comment necessary

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherHourlyRecyclerViewAdapter.java [hunk 6]
```diff
@@ -18,12 +18,11 @@
...
     private final List<WeatherData> mWeathers;
-    private final Map<WeatherData, Boolean> mExpandedFlags;
 
-    public WeatherHourlyRecyclerViewAdapter(List<WeatherData> items) {
+    public WeatherHourlyRecyclerViewAdapter(List<WeatherData> items)
+    {
         this.mWeathers = items;
-        mExpandedFlags = mWeathers.stream()
-                .collect(Collectors.toMap(Function.identity(), weather -> false));
+
```
Comment (gtullio12) [inline app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherHourlyRecyclerViewAdapter.java:25]: Unnecessary line

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherHourlyRecyclerViewAdapter.java [hunk 7]
```diff
@@ -61,14 +60,17 @@ public WeatherViewHolder(View view) {
             binding = FragmentWeatherHourlyCardBinding.bind(view);
         }
 
-
-
         /**
-         * Helper used to determine if the preview should be displayed or not.
+         * helps to set the hourly weather components
+         * @param weather
          */
```
Comment (gtullio12) [inline app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherHourlyRecyclerViewAdapter.java:67]: Remove the line so there is no space between the java doc and the function

File: app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherListFragment.java (hunk 3 above)
Comment (gedefaws) [reply app/src/main/java/edu/uw/tcss450/chatapp/ui/weather/WeatherListFragment.java:63]: The app won't navigate to the WeatherMapFragment then

**Comment by gtullio12** [comment]
Looks good. You can merge it in

**Pull request merged**